### ISSUE TITLE
Leaving from `Insert mode` by press `ESC` leads to send an `ALT` in rare case

### ISSUE THREAD

**Issue by konosubakonoakua** [neovim/neovim#27344]
### Problem

As described in the title, it's hard to reproduce.

I have the following mapping.
```lua
keymap("i", "<A-j>", "<esc><cmd>m .+1<cr>==gi", { desc = "Move down" })
keymap("i", "<A-k>", "<esc><cmd>m .-2<cr>==gi", { desc = "Move up" })
```

It happend randomly when I press `<ESC>` in Insert mode, then move the cursor by pressing `j`.
Instead of moving the cursor one line below, it triggered the `<A-j>` keymap.
When this issue happens, I try to unmap the keymap by `iu <A-j>` and the problem is gone.

I think I haven't installed any plugin that will send an `Alt` when `InsertLeave` happens, Any advice?

### Steps to reproduce

Rare case, hard to reproduce.
If you are interested, here's my config: https://github.com/konosubakonoakua/lazyvim.conf

### Expected behavior

Don't send `Alt` when InsertLeave by Pressing `esc`

### Neovim version (nvim -v)

NVIM v0.10.0-dev-2121+g06aebfa99

### Vim (not Nvim) behaves the same?

dont know

### Operating system/version

ubuntu 22

### Terminal name/version

Guake/GnomeTerminal

### $TERM environment variable

xterm-256color

### Installation

build from repo

**Comment by glepnir** [comment]
you should try use a minimal config for test . provide a distro config not useful .

**Comment by konosubakonoakua** [comment]
> you should try use a minimal config for test . provide a distro config not useful .

Yes, but it's hard to reproduce. I encountered this issue about 3 times a day or less in a heavy usage.
I put the issue here for some advice or clue in ahead.
I will try to reproduce the problem by using a minimal config which only contains the keymap.
```lua
keymap("i", "<A-j>", "<esc><cmd>m .+1<cr>==gi", { desc = "Move down" })
keymap("i", "<A-k>", "<esc><cmd>m .-2<cr>==gi", { desc = "Move up" })
```

**Comment by zeertzjq** [comment]
`:h 'ttimeoutlen'`

**Comment by konosubakonoakua** [comment]
> * `:h 'ttimeout'`
> * `:h 'ttimeoutlen'`
> 
> This is a limitation of many terminal emulators.
I do have a `<esc><esc>` keymap in terminal mode, but not in a insert mode.
If with inappropriate `ttimeoutlen` , will `i<esc>` sequence sending a `ALT` ???
Currently I have `ttimeoutlen=50`, could you please give some explaination on the question mentioned above, thank you. @zeertzjq

**Comment by zeertzjq** [comment]
Please search existing issues, line #8213

**Comment by konosubakonoakua** [comment]
Confirmed with the issue by pressing `<ESC>j` really fast under ttimeoutlen=50,
However I remember that the `i<ESC>j` sequence still can be triggered as `<A-j>` even not under ttimeoutlen=50, I can press the sequence that fast when editing.

Thanks for your advice 👍 , and now I set `ttimeoutlen=30`, cannot trigger `<A-j>` by pressing `<ESC>j` even if I almost break my keyboard 🤣.

As for the `i<ESC>j`, I will try to reproduce under such a low timeout.
